include/engine/datafacade/shared_datafacade.hpp

```diff
@@ -276,11 +276,6 @@ class SharedDataFacade final : public BaseDataFacade
 
     void LoadCoreInformation()
     {
-        if (data_layout->num_entries[storage::SharedDataLayout::CORE_MARKER] <= 0)
-        {
-            return;
-        }
-
```

Comment:
This was the bug that was causing the seg fault in the tests. Swapping non-core enabled datasets after core-enabled datasets caused the m_is_core_node vector not to be updated, causing invalid memory access. We could break that out as own PR but it is quite atomic and a requirement anyway.

Since this is quite an edge case that we basically only will run in when we do testing, I don't think this is a critical bug on its own.